Common subdirectories: a/pcs and b/pcs
Common subdirectories: a/pcsd and b/pcsd
diff -nur a/pcs/utils.py b/pcs/utils.py
--- a/pcs/utils.py	Fri Nov  4 15:04:09 2016
+++ b/pcs/utils.py	Fri Nov  4 15:27:56 2016
@@ -901,8 +901,10 @@
         else:
             stdin_pipe = None
 
+        pcsd_dir_path = settings.pcsd_exec_location
         p = subprocess.Popen(
             args,
+            cwd=pcsd_dir_path,
             stdin=stdin_pipe,
             stdout=subprocess.PIPE,
             stderr=(subprocess.PIPE if ignore_stderr else subprocess.STDOUT),
@@ -952,7 +954,7 @@
     gem_home = os.path.join(pcsd_dir_path, 'vendor/bundle/ruby')
     env_var["GEM_HOME"] = gem_home
     output, retval = run(
-        ["/usr/bin/ruby", "-I" + pcsd_dir_path, pcsdcli_path, command],
+        ["/usr/bin/ruby23", "-I" + pcsd_dir_path, pcsdcli_path, command],
         string_for_stdin=json.dumps(data),
         env_extend=env_var
     )
diff -nur a/pcsd/pcsd-cli.rb b/pcsd/pcsd-cli.rb
--- a/pcsd/pcsd-cli.rb	Fri Nov  4 15:09:59 2016
+++ b/pcsd/pcsd-cli.rb	Fri Nov  4 15:10:33 2016
@@ -1,5 +1,8 @@
 #!/usr/bin/ruby
 
+$LOAD_PATH.unshift File.dirname(__FILE__)
+
+require 'bundler/setup'
 require 'rubygems'
 require 'etc'
 require 'json'
